client/admin/src/views/Orders.vue

```diff
@@ -0,0 +1,233 @@
+import { transform } from "@/utils/orders";
+import { cleanupData } from "@/utils/helper";
+import { Paginator } from "@/types/paginator";
+const OrdersAction = namespace("orders", Action);
+const OrdersGetter = namespace("orders", Getter);
+const PaginatorGetter = namespace("paginator", Getter);
+const PaginatorAction = namespace("paginator", Action);
+
+@Component({
+  components: {
+    Table,
+    ErrorNotification,
+    SuccessNotification,
+  },
+})
+export default class Orders extends Vue {
+  // Errors array
+  private errors: string[] = [];
+
+  private showSettings: Boolean = false;
+
+  private orders: object[] = [];
+  private ordersDisplay: object[] = [];
+
+  private ordersAttributes: string[] = [];
+  private tableColums: string[] = [];
+
+  @OrdersAction ordersResults!: any;
+  @OrdersGetter loadingOrdersResults!: Boolean;
+  @PaginatorGetter paginatorInfo!: Paginator;
+  @PaginatorAction updatePaginatorInfo!: any;
+
+  private paginatorState: Paginator = this.$store.state["paginator"];
+
+  public resetErrors(): void {
+    this.errors = [];
+  }
+
+  // Watch when table component changes the pagination state, then update orders
+  @Watch("paginatorState", { deep: true })
+  onPaginatorStateChanged(state: { reload: Boolean }) {
+    if (state.reload == true) {
+      this.loadOrders();
+    }
+  }
+
+  public paginatorResultsPerPageInputChangeHandle() {
+    this.loadOrders();
+  }
+
+  public checked(event: any): void {
+    let checked = event.target.checked;
+    let value = event.target.value;
+
+    /**
+     * Based on whether the attribute of the order is selected in the settings panel,
+     * add or remove the order attribute from the tableColums array
+     */
+
+    if (checked) {
+      this.tableColums.push(value);
+    } else {
+      var index = this.tableColums.indexOf(value);
+      if (index > -1) {
+        this.tableColums.splice(index, 1);
+      }
+    }
+
+    this.filterOrders();
+  }
+
+  async mounted() {
+    this.loadOrders();
+  }
+
+  public filterOrders() {
+    /**
+     * I have an original array (this.orders) and another array (this.ordersDisplay) , the orders array will never be
+     * changed, only the selected values from the settings panel will be used to filter data from the original orders array
+     * back into the ordersDisplay array and then appear in the ui to the user.
+     *
+     * The orders array will keep all the data from graphql query that returns the order attributes. The ordersDisplay will change
+     * each time the user selects more columns or removes columns from the setttings tab. Each column value is a property in the poduct
+     * object. So for each order object in the orders array we filter out only the desired keys from the objects and clone them
+     * into the ordersDisplay array.
+     */
+    let tableColums = this.tableColums;
+
+    this.ordersDisplay = this.orders.map(function (order: any) {
```

Comment:
@AlexeiaW just wondering if we should move this into a helper as well, since it looks like quite a generic function, what do you think?

Reply:
Ok yeah we can do that, makes sense